client/src/store/modules/user.js

```diff
@@ -28,7 +28,12 @@ export default {
         const user = await firebase
           .auth()
           .createUserWithEmailAndPassword(payload.email, payload.password);
-        const userPayload = { id: user.user.uid };
+        if(payload.username != ""){
```

Comment:
why do we need tat part?

Reply:
since its username field is kept optional. I thought it would be good if we don't make the unnecessary call to firebase to store username if it isn't defined by user

Reply:
I just realised that we don't need to store it on firebase at all:) firebase is used here only for authentication

Reply:
okay got it

Reply:
also, I noticed that we have a post request again into our MongoDB after signing in, which IMO is not required. we need to store user info only once in MongoDB during signUp.

Reply:
good catch!

Reply:
If you want to gain more PRs for Hacktoberfest t-shit. You can create another PR for that issue, I'll merge both of them

Reply:
Thank you. please merge this one. Then I commit that change and raise the new pull request otherwise I observed the commits get auto added to this existing pr